fabric-recipe-api-v1/src/main/java/net/fabricmc/fabric/mixin/recipe/RecipeManagerMixin.java

```diff
@@ -55,22 +55,21 @@ private void onReload(Map<Identifier, JsonElement> map, ResourceManager resource
 		RecipeManagerImpl.apply(map, builderMap);
 	}
 
-	// Use an inject instead of a redirect. A redirect is technically impossible as it will not allow a
-	// Map<Identifier, Recipe<?>> as return type.
-	@Inject(
-			method = "method_20703", // synthetic method in the toImmutableMap in the apply method.
-			at = @At(value = "HEAD"),
-			cancellable = true
-	)
-	private static void onImmutableMapBuilder(Map.Entry<RecipeType<?>, ImmutableMap.Builder<Identifier, Recipe<?>>> entry,
-									CallbackInfoReturnable<Map<Identifier, Recipe<?>>> cir) {
+	/**
+	 * Synthetic method in {@link RecipeManager#apply(Map, ResourceManager, Profiler)} as an argument of {@code toImmutableMap}.
+	 *
+	 * @author Fabric API
+	 * @reason Replaces immutable maps for mutable maps instead.
+	 */
+	@Overwrite
+	private static Map<Identifier, Recipe<?>> method_20703(Map.Entry<RecipeType<?>, ImmutableMap.Builder<Identifier, Recipe<?>>> entry) {
 		// This is cursed. Do not look.
-		cir.setReturnValue(ImmutableMapBuilderUtil.specialBuild(entry.getValue()));
+		return ImmutableMapBuilderUtil.specialBuild(entry.getValue());
 	}
 
 	@Inject(
 			method = "apply",
-			at = @At(value = "INVOKE", target = "Lorg/apache/logging/log4j/Logger;info(Ljava/lang/String;Ljava/lang/Object;)V")
+			at = @At(value = "INVOKE", target = "Lorg/apache/logging/log4j/Logger;info(Ljava/lang/String;Ljava/lang/Object;)V", remap = false)
```

Comment:
remap = false is typically a bad omen, please make sure that works in production.

Reply:
it disables remapping of the Logger#info

Without it it logs a warning when compiling about not finding any mappings for it anyway.

Reply:
This is a library call, no need for remap lol